packages/website/blog/2022-12-23-consistent-type-exports-and-imports-why-and-how.md

```diff
@@ -0,0 +1,97 @@
+---
+authors:
+  - image_url: https://www.joshuakgoldberg.com/img/josh.jpg
+    name: Josh Goldberg
+    title: typescript-eslint Maintainer
+    url: https://github.com/JoshuaKGoldberg
+description: Why enforcing TypeScript imports use the `type` modifier when possible benefits some project setups.
+slug: consistent-type-exports-and-imports-why-and-how
+tags: [typescript, imports, exports, types, transpiling]
+title: 'Consistent Type Exports and Imports: Why and How'
+---
+
+TypeScript 3.8 [added type-only imports and exports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html):
+
+```ts
+import type { SomeThing } from './some-module.js';
+export type { SomeThing };
+```
+
+The key difference with `export type` and `import type` is that they are _only in the type system_.
+Attempting to use a _value_ imported as only a _type_ in runtime code will cause a TypeScript error:
+
+```ts twoslash
+import type { SomeThing } from './some-module.js';
+
+new SomeThing();
+//  ~~~~~~~~~
+// 'SomeThing' cannot be used as a value
+// because it was imported using 'import type'.
+```
+
+This brings up two questions:
+
+- Why would you want to use these type-only imports and exports?
+- How can you enforce a project use them whenever necessary?
+
+Let's Dig In!
+
+<!--truncate-->
+
+## Functional Benefits
+
+Some modules in code may cause _side effects_: code that is run when the module is imported and causes changes outside the module.
+Common examples of side effects include calling to global APIs like `fetch` or creating DOM stylesheets.
+
+When projects include modules that cause side effects, the order of module imports matters.
+For example, some projects import the types of side-effect-causing modules in code that needs to run before those side effects.
+
+Import statements that only import types are generally removed when the built-in TypeScript compiler transpiles TypeScript syntax to JavaScript syntax.
+The built-in TypeScript compiler is able to do so because it includes a type checker that knows which imports are of types and/or values.
+
+But, some projects use transpilers such as Babel and Vite that don't have access to type information.
```

Comment:
```suggestion
But, some projects use transpilers such as Babel and Vite that don't have access to type information.
These transpilers are often referred to as isolated module transpilers because they effectively transpile each module in isolation from other modules.
```

possible extra context sentence?